home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/invitations/InvitationApiIT.java

```diff
@@ -31,12 +48,122 @@ void isEmailSentTest() throws Exception {
         assertTrue(response.getCount() > 0);
     }
 
+    @Test
+    void isApartmentTokenAccepted() throws Exception {
+        String userEmail = RandomStringUtils.randomAlphabetic(10).concat("@gmail.com");
+        ReadCooperation readCooperation = createTestCooperationAndUserViaInvitationWithUserEmail(userEmail);
+
+        ReadHouse createdHouse = houseApi.createHouse(readCooperation.getId(), createHouse());
+        apartmentApi.createApartment(createdHouse.getId(), createApartmentWithEmail(userEmail));
+
+        TimeUnit.MILLISECONDS.sleep(5000);
+
+        ApiUsageFacade api = new ApiUsageFacade();
+        MailHogApiResponse mailResponse = api.getMessages(new ApiMailHogUtil(), MailHogApiResponse.class);
+
+        String apartmentInvitationToken = getToken(getDecodedMessageByEmail(mailResponse, userEmail));
+
+        ReadInvitation readInvitation = invitationApi.approveInvitation(buildInvitationPayload(apartmentInvitationToken));
+
+        assertEquals(userEmail, readInvitation.getEmail());
+        assertEquals(InvitationType.APARTMENT, readInvitation.getType());
+        assertEquals(InvitationStatus.ACCEPTED,readInvitation.getStatus());
+    }
+
+    @Test
+    void isCooperationTokenAccepted() throws Exception {
+        String userEmail = RandomStringUtils.randomAlphabetic(10).concat("@gmail.com");
+        createTestCooperationAndUserViaInvitationWithUserEmail(userEmail);
+        createTestCooperationWithUserEmail(userEmail);
+
```

Comment:
What for to create two cooperations with same admin email in this test? To check that two invitations came to the same email? That both UserCooperations registered? But this is not verified in the test.

Reply:
We need it to check invitation registration of a user to second cooperation as admin. Because invitation to already registred user has different flow.